### PR TITLE
Upgrade docker images in dockerfile

### PR DESCRIPTION
**Pull request by alpe**
* Fixes failing docker build
* Enables docker build for all branches

### CODE REVIEW

**Comment by ethanfrey** [comment]
Looking at the build failure... 

"Setup a remote docker engine"

```
Server Engine Details:
  Version:          17.09.0-ce
  API version:      1.32 (minimum version 1.12)
  Go version:       go1.8.3
  Git commit:       afdb6d4
  Built:            2017-09-26T22:40:56.000000000+00:00
  OS/Arch:          linux/amd64
  Experimental:     false
```

Think this is a version issue?

**Comment by ethanfrey** [comment]
Bisecting when docker broke in master, I found this:

Screenshot 2022-01-25 at 13 22 30

**Comment by ethanfrey** [comment]
The last working one had the same docker engine

**Comment by alpe** [comment]
The "issue" was introduced with the alpine image upgrade. The remote docker version became incompatible. See https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.14.0#faccessat2 for details

File: .circleci/config.yml [hunk 1]
```diff
@@ -182,10 +186,10 @@ workflows:
       - docker-image:
           requires:
             - setup-dependencies
-          filters:
-            branches:
-              only:
-                - master
+#          filters:
```
Comment (ethanfrey) [inline .circleci/config.yml:189]: We need these back!

**Review by ethanfrey — COMMENTED** [review]
I think you merged without the filters.

We should get those back (one more PR before tagging)

**Pull request merged**